File: frontend/src/types/aiUsage.ts
/**
 * AI usage / cost telemetry contracts.
 *
 * Mirrors backend/app/schemas/ai_usage.py (GET /api/v1/ai-usage/summary):
 * read-only aggregates over the AI usage ledger for the active company.
 */

/** Aggregated usage for one bucket (a task, a model, or the whole window). */
export interface AIUsageAggregate {
  calls: number;
  input_tokens: number;
  output_tokens: number;
  cache_creation_tokens: number;
  cache_read_tokens: number;
  /** Estimated spend in USD; null when no priced calls landed in the bucket. */
  estimated_cost_usd: number | null;
  avg_latency_ms: number | null;
  /** Failed calls / total calls, 0.0-1.0. */
  error_rate: number;
}

export interface AIUsageTaskSummary extends AIUsageAggregate {
  task: string;
}

export interface AIUsageModelSummary extends AIUsageAggregate {
  model: string;
}

export interface AIUsageSummaryResponse {
  window_days: number;
  /** ISO datetime — start of the aggregation window. */
  since: string;
  totals: AIUsageAggregate;
  by_task: AIUsageTaskSummary[];
  by_model: AIUsageModelSummary[];
}

File: backend/app/schemas/ai_usage.py
"""Response contracts for the AI usage telemetry endpoints."""

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel

from app.schemas.base import UTCModel


class AIUsageAggregate(BaseModel):
    """Aggregated usage for one bucket (a task, a model, or the whole window)."""

    calls: int
    input_tokens: int
    output_tokens: int
    cache_creation_tokens: int
    cache_read_tokens: int
    estimated_cost_usd: Optional[float] = None  # None when no priced calls in the bucket
    avg_latency_ms: Optional[float] = None
    error_rate: float  # failed calls / total calls, 0.0-1.0


class AIUsageTaskSummary(AIUsageAggregate):
    task: str


class AIUsageModelSummary(AIUsageAggregate):
    model: str


class AIUsageSummaryResponse(UTCModel):
    window_days: int
    since: datetime
    totals: AIUsageAggregate
    by_task: List[AIUsageTaskSummary]
    by_model: List[AIUsageModelSummary]
